datasource/src/main/scala/database/SlickDatabase.scala

```diff
@@ -0,0 +1,58 @@
+package database
+
+import java.sql.Connection
+
+import com.typesafe.config.{Config, ConfigFactory}
+import liquibase.database.DatabaseFactory
+import liquibase.database.jvm.JdbcConnection
+import liquibase.resource.ClassLoaderResourceAccessor
+import liquibase.{Contexts, LabelExpression, Liquibase}
+import slick.basic.DatabaseConfig
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+
+class SlickDatabase(val config: Config = ConfigFactory.load()) extends SqlDatabase with AutoCloseable {
+
+  private val slickConfig = DatabaseConfig.forConfig[JdbcProfile](path = "postgres_dc", config)
+
+  protected val profile = slickConfig.profile
+  protected val database = slickConfig.db
+
+  import slickConfig.profile.api._
+
+  private def withConnection[A](actionFunc: Connection => A): A = {
+    val action = SimpleDBIO(context => actionFunc(context.connection))
+    Await.result(database.run(action), Duration.Inf)
```

Comment:
Is it possible to avoid using of the blocking call of 'Await'? Is it critical here?

Reply:
This method is used for DB migration which should be performed in a blocking way before all other app's components will start. If we look under this angle - it makes sense.